File: clibs/libgpio.c
#include<stdio.h>
#include<pigpio.h>

int initGPIO(){
	return gpioInitialise();
}

void termGPIO(){
	printf("\nTerminating GPIO lib\n");
	gpioTerminate();
}

int readGPIO(unsigned GPIOnumber){
	return gpioRead(GPIOnumber);
}

int writeGPIO(unsigned GPIOnumber, unsigned level){
	return gpioWrite(GPIOnumber, level);
}

int pwmGPIO(unsigned GPIOnumber, unsigned power){
	return gpioPWM(GPIOnumber, power);
}

int resetPwmGPIO(unsigned GPIOnumber){
	return gpioPWM(GPIOnumber, 0);
}

File: clibs/libgpio.js
var ffi = require('ffi');

var libgpio = ffi.Library('./clibs/libgpio_smoke', {
	'initGPIO': [ 'int', []],
	'termGPIO': [ 'void', []],
	'readGPIO': [ 'int', [ 'int' ]],
	'writeGPIO': [ 'int', [ 'int', 'int' ]],
	'pwmGPIO': [ 'int', [ 'int', 'int' ]],
	'resetPwmGPIO': [ 'int', [ 'int' ]]
});

exports.init = libgpio.initGPIO;
exports.term = libgpio.termGPIO;
exports.read = libgpio.readGPIO;
exports.write = libgpio.writeGPIO;
exports.pwm = libgpio.pwmGPIO;
exports.reset = libgpio.resetPwmGPIO;

